viewing_party/party.py

```diff
@@ -1,23 +1,178 @@
-# ------------- WAVE 1 --------------------
-
+# 1st function in Wave 1
 def create_movie(title, genre, rating):
-    pass
 
-# -----------------------------------------
-# ------------- WAVE 2 --------------------
-# -----------------------------------------
+    if title is None or genre is None or rating is None:
+        return None
+
+    new_movie = {"title": title, "genre": genre, "rating": rating}
+
+    if isinstance(title, str) or isinstance(genre, str) or isinstance(rating, int):
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+        }
+    else:
+        return None
+
+
+# 2nd function in Wave 1
+def add_to_watched(user_data, movie):
+
+    (user_data["watched"]).append(movie)
+    return user_data
+
+
+# 3rd function in Wave 1
+def add_to_watchlist(user_data, movie):
+
+    watchlist = user_data.get("watchlist", [])
+
+    watchlist.append(movie)
+    user_data["watchlist"] = watchlist
+    
+    return user_data
+
+
+# 4th function in Wave 1
+def watch_movie(user_data, title):
+
+    move_movie = False
+    index = 0
+    movie = None
+
+    for index in range(len(user_data["watchlist"])):
+        if title == user_data["watchlist"][index]["title"]:
+            move_movie = True
+            movie_index = index
+            movie = user_data["watchlist"][index]
+            print(user_data["watchlist"][index])
+    if move_movie:
+        (user_data["watched"]).append(movie)
+        del user_data["watchlist"][movie_index]

```

Comment:
I see that this line deletes the item in `user_data["watchlist"]` at `movie_index`! Your approach uses the `del` keyword, which is a good popular solution. For me, I always think my code looks/feels weird when I have to use a Python keyword, so I want to provide the alternative `pop`:

```python
        user_data["watchlist"].pop(movie_index)
```

The tests still pass!